### PR TITLE
feat(ccp): Prometheus metrics for I/O

### PR DESCRIPTION
**Pull request by monoid**
Add following metrics:

 + `ccp_io_error_total` counter
 + `ccp_writing_time_seconds_*` histogram

Both metrics have two labels: `storage=state` and `storage=proof`.

### CODE REVIEW

File: ccp/src/persistence.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    // we might use random name here.  but if the dir will lock'd, it is not necessary.
+    let base_dir = path.parent().map(Cow::Borrowed).unwrap_or_default();
+    let base_dir_file = File::open(&base_dir)?;
+
+    let (mut draft_file, draft_path) =
+        gen_draft_file(&base_dir, path.file_name().unwrap_or_default())?;
+
+    draft_file.write_all(contents.as_ref())?;
+    draft_file.flush()?;
+    draft_file.sync_all()?;
```
Comment (folex) [inline ccp/src/persistence.rs:81]: It's usually nice to add `path` to the error context, wdyt?

File: ccp/src/persistence.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    std::mem::drop(draft_file);
+
+    // now we have the contents saved reliably on disk as a draft file.
+    // if we rename, destination file will be changed atomically
```
Comment (folex) [inline ccp/src/persistence.rs:85]: ❤️

**Review by folex — APPROVED** [review]
well done!

File: ccp/src/persistence.rs (hunk 1 above)
Comment (monoid) [reply ccp/src/persistence.rs:81]: It was done in #119, still on review.

File: ccp/src/persistence.rs (hunk 2 above)
Comment (monoid) [reply ccp/src/persistence.rs:85]: Thank you very much, but this is an old code moved from other file.

File: ccp/src/hashrate/prometheus.rs [hunk 3]
```diff
@@ -57,6 +62,38 @@ async fn handle_metrics(
             guard.apply_to_registry(&mut registry);
         }
 
+        {
```
Comment (mikevoronov) [inline ccp/src/hashrate/prometheus.rs:65]: What is the purpose of this scope here?

Comment (monoid) [reply ccp/src/hashrate/prometheus.rs:65]: Just to visually separate state storage and proof storage stats handling.
Perhaps, commenting would be better.

**Pull request merged**